### PR TITLE
Downgrade to avatica-server 1.8.0, skip avatica-core.

### LINKED ISSUE #3800: Jackson version conflict with avatica
There is a conflict with jackson-databind 2.6.3 (from transitive dependencies of calcite-core and avatica-core) and druid's explicit jackson-databind 2.4.6 dependency. I see issues during batch ingestion. Realtime using tranquility is fine. Here is a gist containing the stack trace

Druid has direct dependency on `calcite-core`, `avatica-core` and `avatica-server` from https://github.com/druid-io/druid/pull/3682.

`avatica-core` is the non-shaded artifact which only contains client classes without any bundled dependencies except jackson. as per this: https://github.com/apache/calcite/commit/e5a42f1ca80eca0bb6f0c8f6677f106fadb2e652#diff-eb4d33b91f63eeaaea2ffd36688b61f8R49

`calcite-core` depends on shaded avatica.
`avatica-server` depends on non-shaded avatica
`avatica-core` is the non-shaded avatica.

cc: @gianm

### PR DESCRIPTION
**Pull request by gianm**
This matches the version bundled by Calcite 1.10.0.

Attempts to fix #3800.

### CODE REVIEW

**Comment by gianm** [comment]
Please hold off on merging until @praveev can confirm this actually fixes the reported issue #3800.

**Comment by praveev** [comment]
@gianm I can still repro #3800. Can you also try a batch ingestion job and confirm that you can repro? If it helps, I've posed a qn. to @joshelser who addressed this with avatica 1.9.0. https://issues.apache.org/jira/browse/CALCITE-1110

**Comment by joshelser** [comment]
Just posted a reply at https://issues.apache.org/jira/browse/CALCITE-1110?focusedCommentId=15775301&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15775301.

tl;dr Create your own Maven module which bundles Avatica, but shades its dependencies, include this artifact as a dependency and exclude the upstream Apache Avatica dependency instead.

Will keep an eye out here and try to describe better if this is not clear.

**Comment by gianm** [comment]
I looked into this a bit more and it looks like the problem is that Calcite 1.10.0 depends on avatica-1.8.0 which includes non-relocated Jackson classes. This is bad because those classes will get loaded by all of Druid instead of the Jackson classes we actually want. Calcite master (not yet released) depends on avatica-core-1.9.0, which does not include Jackson classes, so it shouldn't have this problem. But it hasn't been released yet.

So some possible solutions:

1. Include avatica-1.8.0 in the assembly but remove the Jackson classes. Maybe it's possible to do this with some fancy Maven footwork (something involving `maven-dependency-plugin:unpack` in the distribution module)?
2. Create our own version of avatica as @joshelser suggests (I guess this means druid-avatica).
3. Wait for Calcite 1.11.0 to be released, this appears imminent and likely will be before the next Druid release (https://mail-archives.apache.org/mod_mbox/calcite-dev/201612.mbox/browser).
4. Excluding avatica-1.8.0 and upgrading to avatica-core-1.9.0 doesn't work, since Calcite 1.10.0 won't run properly against avatica-core-1.9.0 due to some method signature changes.

I'm leaning towards (3) since that looks simplest, but if time is critical, I would be okay with (1) or (2) as well. (1) and (2) are both hacky but (1) seems better and less hacky to me.

@praveev what are your thoughts?

**Comment by gianm** [comment]
@joshelser, thanks for the suggestion. It seems to me this should be a lot simpler to deal with when Calcite 1.11.0 is out; does that sound right to you?

**Comment by praveev** [comment]
@gianm 👍 to wait for calcite 1.11.0. looks like they plan to release in early Jan. https://issues.apache.org/jira/browse/CALCITE-1547

**Comment by joshelser** [comment]
> It seems to me this should be a lot simpler to deal with when Calcite 1.11.0 is out; does that sound right to you?

Yeah, this is just the inherent lag between Avatica and Calcite releases lining up. In the future, I'd like to solidify API between the two projects which should let you do things like just provide a newer version of Avatica with Calcite, not worrying about method signature changes.

I'll try to keep it in mind when we start getting RC's together for Calcite 1.11.0 and mention it again to make sure that we revisit this issue before it's released (not that we'd have to turn around and make a 1.11.1).

**Comment by gianm** [comment]
Thanks @joshelser and @praveev. In that case I'll close this PR and we can do a new one under #3800 when Calcite 1.11.0 is released.
